test/python/algorithms/test_ipe.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test iterative phase estimation"""
+
+import math
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from ddt import ddt, data, unpack
+from qiskit.algorithms.phase_estimators import IterativePhaseEstimation
+import qiskit
+from qiskit import QuantumCircuit
+from qiskit.opflow import (H, X, Z, I)
+
+
+@ddt
+class TestIterativePhaseEstimation(QiskitAlgorithmsTestCase):
```

Comment:
Would it make sense to merge this with the PE tests? It looks like they are the same except you swap out the classes. For Amplitude Estimation we have that same structure, you could check there to compare 🙂

Reply:
That makes some sense. TLDR: I'll try it.
OTOH there are two viewpoints: 1. "refactor relentlessly". And (something you can see whole talks on) 2. Don't refactor too early. Because, sometimes you want to add a class or change something, in which case you can add workarounds making your refactored design more brittle, or else un-refactor ("expand"?). In the current case the amount of code is small enough that it doesn't make much difference either way.

Reply:
Done